### DISCUSSION TITLE
Add Keepassxc to the Microsoft Store

### DISCUSSION THREAD

**Discussion by thewalkerz** [keepassxreboot/keepassxc#7188] [Ideas]
Some time ago Microsoft released its  new application store so that more developers could upload their apps, regardless if they are UWP, PWA or the classic Win32, I would like them to upload Keepassxc to the Microsoft store, that would make it easier to download this fabulous password manager.

This video from Microsoft explains how to do with Win32
https://www.microsoft.com/en-us/videoplayer/embed/RWFAsa?pid=cx-video-1-oneplayer&postJsllMsg=true&autoplay=true&market=en-us

Thanks for this Amazing Password Manager 😁

**Comment by droidmonkey** [comment]
Yup, planning on doing this for the 2.7.0 release coming up! Thanks for the kudos 👏

**Reply by unicorn855** [reply]
Is this still planned/likely to happen? I am currently thinking about whether or not to switch out of 'S Mode' on one of windows devices because I'd rather have a great password manager than the 'increased security' which windows in S mode claims to offer :)
Still, I'd like to see a robust and feature-rich password manager such as KeepassXC on there regardless 👍

**Comment by droidmonkey** [comment]
I am working on getting registered to add an app to the store... a bit more cumbersome than I expected.

**Reply by unicorn855** [reply]
Thank you for the heads-up :-) good to know it's happening regardless of when it does I think. 🤞🤞

**Reply by hj-collab** [reply]
@droidmonkey Any update on the MS Store availability?

**Reply by droidmonkey** [reply]
I can't get there dumb partner center to let me add an app.... I am really not sure how ANY apps are loaded in their store, it is absolutely terrible experience.

**Reply by hj-collab** [reply]
@droidmonkey I have no idea of the process. May be you can ask for help from another open source maintainer who have ms store delivery mechanism? https://github.com/M2Team/NanaZip

**Reply by droidmonkey** [reply]
Good news, I was able to get Microsoft to reset my account and I have submitted our application for publication. Once Microsoft confirms the submission it will be live on the store!

**Reply by hj-collab** [reply]
@droidmonkey Any idea how many days it takes for the application to get approved? I was searching in the store but couldn't find KeePassXC yet.

**Reply by droidmonkey** [reply]
They keep rejecting the app for non sensicle reasons. I don't have time to constantly try to submit this thing.

**Reply by hj-collab** [reply]
@droidmonkey I wish some body could help you with the process. I am disappointed to see the shitty password managers and bad keepass clones on the store. It really needs a software like yours. I hope some body who knows the process well offers to help. I still suggest taking help from NanaZip team. Their niche is different but still they should be able to help with you.

**Reply by hj-collab** [reply]
@droidmonkey Now it's really important for the official app to get into the store. Else fake keepassxcs will make money in your name and might even sniff passwords. I suggest you to ask for help in Privacy Communities like PrivacyGuides to get the app in the store.

**Reply by droidmonkey** [reply]
Just because we are in the store already doesn't prevent anyone else from submitting. Microsoft only cares if the name is the same, not the actual application. 

I am attempting another submission right now.

**Reply by hj-collab** [reply]
@droidmonkey All the best. Meanwhile I'll ask people in the PrivacyGuides to report the fake app.

**Reply by unicorn855** [reply]
I mean it's a dead giveaway that the app in the store at the moment is fake not only because no one knows who/what 'CoderLearn' is supposed to be, but also because they literally infringe upon your rights by claiming to be the 'KeePassXC Team' (what a dodgy phrasing to begin with)

Can I as a simple user do anything about that scam in any case?

Perhaps it's a good idea to issue a warning about the fake app on your official website until this gets resolved?

**Reply by droidmonkey** [reply]
We have issued a formal copyright takedown request and reached out to Microsoft.

**Comment by droidmonkey** [comment]
Anddddddd DONE

https://apps.microsoft.com/store/detail/keepassxc/XP8K2L36VP0QMB

**Reply by hj-collab** [reply]
@droidmonkey Not happy with the fact that the ms store release have unsigned dll files.

![image](https://user-images.githubusercontent.com/101234934/194032667-3568bd2d-4dfb-4e37-89e0-a5bc08569dae.png)

It does not work with WDAC.

**Reply by hj-collab** [reply]
@droidmonkey Is it also possible to drop UAC escalation for installation/update?

**Reply by droidmonkey** [reply]
I just rebuilt vcpkg and forgot to sign everything. You can setup WDAC and AppLocker to be less restrictive then everything needs to be signed. Very few applications actually sign all their dlls, especially the bundled ones.

**Reply by hj-collab** [reply]
@droidmonkey I have shifted back to the msi package on website from MS store package. I run WDAC very strict configuration and it works fine with the apps I use. I don't want to tone it down. Thanks!

**Reply by zmatthias** [reply]
It would be really aweseome if UAC escalation were not needed. Me and many people would love to use it for work and sadly do not get the privledges on our machines. Thanks a lot!

**Reply by droidmonkey** [reply]
Use our portable distribution or ask your IT staff to make it available in your software store. Installing a password manager to user modifiable location is very risky. Files could be tampered with that corrupts our program and exposes your secrets.

**Reply by hj-collab** [reply]
@droidmonkey I'll use the portable distribution. Just wanted to know your thoughts. Is it really risky when all your KeePassXC binaries & dlls are signed? It's not some body could easily tamper them.

**Reply by droidmonkey** [reply]
If you use applocker and enforce binary signing then you are better off.

**Comment by Vettenranta** [comment]
If I download KeepassXC from Microsoft store does it need to be verified like the .msi and .zip files? I mean the authenticity and integrity checks. I am very paranoid about my security.

**Reply by phoerious** [reply]
If you downloaded this, no.
https://apps.microsoft.com/store/detail/keepassxc/XP8K2L36VP0QMB

**Reply by Vettenranta** [reply]
Yes, that's the one that I downloaded.

**Comment by wbrickles** [comment]
It appears this app doesn't update from the Microsoft Store (I understand that's a limitation of win32 apps).  There also doesn't be the standard way of checking for updates from the app itself.  What is the recommended way for Microsoft Store users to keep the application updated?

**Reply by phoerious** [reply]
Microsoft takes forever to approve new updates.

**Reply by droidmonkey** [reply]
So we learned that win32 apps are NOT auto updated from the store, which is so dumb. You need to install the update outside the store.

**Reply by wbrickles** [reply]
My store version (2.7.4) doesn't appear to have "Check for updates" available as an option.  So is the best option for updating to download from the website?

**Reply by droidmonkey** [reply]
Yes

**Reply by wbrickles** [reply]
That's unfortunate, really leaves people who install from the Microsoft Store pretty vulnerable if any security issue is found.  Thank you for the quick responses!
